apps/deploy-web/src/components/new-deployment/NewDeploymentContainer.tsx

```diff
@@ -111,13 +111,22 @@ export function NewDeploymentContainer() {
     }
   }
 
+  function onTemplateSelected(template: TemplateCreation | null) {
+    setSelectedTemplate(template);
+  }
+
   return (
     <Layout isLoading={isLoadingTemplates} isUsingSettings isUsingWallet containerClassName="pb-0">
       <div className="flex w-full items-center">{activeStep !== null && <CustomizedSteppers activeStep={activeStep} />}</div>
 
-      {activeStep === 0 && <TemplateList setSelectedTemplate={setSelectedTemplate} setEditedManifest={setEditedManifest} />}
+      {activeStep === 0 && <TemplateList />}
       {activeStep === 1 && (
-        <ManifestEdit selectedTemplate={selectedTemplate as TemplateCreation} editedManifest={editedManifest as string} setEditedManifest={setEditedManifest} />
+        <ManifestEdit
+          selectedTemplate={selectedTemplate}
+          onTemplateSelected={onTemplateSelected}
```

Comment:
Again, just for visibility
```suggestion
          onTemplateSelected={setSelectedTemplate}
```